rest/multipart.go

```diff
@@ -27,30 +27,23 @@ const kMaxInlineAttachmentSize = 200
 // JSON bodies smaller than this won't be GZip-encoded.
 const kMinCompressedJSONSize = 300
 
-// Parses a JSON MIME body, unmarshaling it into "into".
+// ReadConfigJSON parses a JSON body, unmarshals it into the target structure "into".
 // Closes the input io.ReadCloser once done.
-func ReadJSONFromMIME(headers http.Header, input io.ReadCloser, into interface{}) error {
-	contentType := headers.Get("Content-Type")
-	if contentType != "" && !strings.HasPrefix(contentType, "application/json") {
-		return base.HTTPErrorf(http.StatusUnsupportedMediaType, "Invalid content type %s", contentType)
+func ReadConfigJSON(headers http.Header, input io.ReadCloser, into interface{}) error {
+	// Performs the Content-Type validation and Content-Encoding check.
+	input, err := processContentEncoding(headers, input)
+	if err != nil {
+		return err
 	}
 
-	switch headers.Get("Content-Encoding") {
-	case "gzip":
-		var err error
-		if input, err = gzip.NewReader(input); err != nil {
-			return err
-		}
-	case "":
-		break
-	default:
-		return base.HTTPErrorf(http.StatusUnsupportedMediaType, "Unsupported Content-Encoding; use gzip")
+	content, err := ioutil.ReadAll(input)
```

Comment:
We're reading the buffer just to create a new one in decode. Seems unnessesary overhead.

Reply:
This has been removed from ReadJSONFromMIME.